OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/pdf/PDFDisplay.java

```diff
@@ -36,9 +36,13 @@ public JPanel drawPreviewOnJPanel() {
         Image generatedImage = PDFPreviewBuilder.buildPDFPreview(selectedPage, viewZoom);
         
         if (generatedImage != null) {
-            ImageIcon icon = new ImageIcon(generatedImage);
-            JLabel picLabel = new JLabel(icon);
-            pdfImagePanel.add(picLabel);
+            try {
+                ImageIcon icon = new ImageIcon(generatedImage);
+                JLabel picLabel = new JLabel(icon);
+                pdfImagePanel.add(picLabel);
+            } catch (OutOfMemoryError oome) {
```

Comment:
The https://maven.apache.org/plugins/maven-compiler-plugin/examples/compile-with-memory-enhancements.html code doesn't work directly, I imagine there is some code elsewhere that needs to be included to activate it.  Will study it more tomorrow.